client/app/services/SettingsMenu.js

```diff
@@ -0,0 +1,40 @@
+import { isFunction, extend, omit, sortBy } from 'lodash';
+
+class SettingsMenuItem {
+  constructor(menuItem) {
+    extend(
+      this,
+      { pathPrefix: `/${menuItem.path}` },
+      omit(menuItem, ['isActive']),
+    );
+    if (isFunction(menuItem.isActive)) {
+      this.isActive = menuItem.isActive;
+    }
+  }
+
+  isActive(path) {
+    return path.startsWith(this.pathPrefix);
+  }
+}
+
+class SettingsMenu {
+  constructor() {
+    this._items = [];
+    this._dirty = false;
+  }
+
+  get items() {
+    if (this._dirty) {
+      this._items = sortBy(this._items, 'order');
```

Comment:
Why not sort when adding a new item and just use `this.items`? Feels like a premature optimization a bit.

Reply:
Yes, I agree. But - why not? This service is very small and probably will not grow at all. I just feel a bit uncomfortable when I see a code that sort something when new items added :slightly_smiling_face: